sycl/include/sycl/detail/pi.hpp

```diff
@@ -18,7 +18,10 @@
 #include <sycl/detail/os_util.hpp> // for __SYCL_RT_OS_LINUX
 #include <sycl/detail/pi.h>        // for piContextCreate, piContextGetInfo
 
-#include <cstdint>     // for uint64_t, uint32_t
+#include <cstdint> // for uint64_t, uint32_t
+#if _WIN32
+#include <filesystem>
+#endif
```

Comment:
Why `if`? `filesystem` is the standard header, so it should work on any platform.

Reply:
Removed include under ifdef by moving code to os specific source. Added todo to refactor code when <filesystem> is available everywhere.

Reply:
Thanks!